--- ../lib/svg-salamander.src/main/java/com/kitfox/svg/Group.java	2012-01-09 23:41:46.000000000 +0100
+++ com/kitfox/svg/Group.java	2012-05-06 17:08:21.000000000 +0200
@@ -203,6 +203,10 @@
                     }
                 }
 
+                if (g instanceof SVGGraphics2D) {
+                	((SVGGraphics2D) g).startRendering(rendEle);
+                }
+                
                 rendEle.render(g);
             }
         }
--- ../lib/svg-salamander.src/main/java/com/kitfox/svg/SVGGraphics2D.java	1970-01-01 01:00:00.000000000 +0100
+++ com/kitfox/svg/SVGGraphics2D.java	2012-05-06 18:57:53.000000000 +0200
@@ -0,0 +1,20 @@
+package com.kitfox.svg;
+
+/**
+ * Interface that can be implemented by the Graphics2D object
+ * passed to SVGDiagram.render() 
+ * 
+ * @author Flemming Frandsen <dren.dk@gmail.com> <http://dren.dk>
+ */
+public interface SVGGraphics2D {
+	/**
+	 * Notifies the Graphics2D target about where the graphics calls come from
+	 * in the SVG document.
+	 * 
+	 * This feature can be used to extract more information from the source svg
+	 * element than what is available via the normal Graphics2D interface.
+	 *  
+	 * @param element the next element that's going to be rendered
+	 */
+	void startRendering(RenderableElement element);
+}
